src/visidentifier/visidentifier.cpp

```diff
@@ -0,0 +1,403 @@
+#include "vismessage.hpp"
+#include "wsexception.hpp"
+
+/***********************************************************************************************************************
+ * VISSubscriptions
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+void VISSubscriptions::RegisterSubscription(const std::string& subscriptionId, Handler&& subscriptionHandler)
+{
+    std::lock_guard lock(mMutex);
+
+    LOG_DBG() << "Registred subscription id = " << subscriptionId.c_str();
+
+    mSubscriptionMap[subscriptionId] = std::move(subscriptionHandler);
+}
+
+aos::Error VISSubscriptions::ProcessSubscription(const std::string& subscriptionId, const Poco::Dynamic::Var value)
+{
+    std::lock_guard lock(mMutex);
+
+    const auto it = mSubscriptionMap.find(subscriptionId);
+
+    if (it == mSubscriptionMap.cend()) {
+        LOG_ERR() << "Unexpected subscription id: = " << subscriptionId.c_str();
+
+        return aos::ErrorEnum::eNotFound;
+    }
+
+    return it->second(value);
+}
+
+/***********************************************************************************************************************
+ * VISIdentifier
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+VISIdentifier::VISIdentifier(const WSConfig& wsConfig)
+    : mWsClientPtr {std::make_shared<PocoWSClient>(
+        wsConfig, std::bind(&VISIdentifier::HandleSubscription, this, std::placeholders::_1))}
+    , mWSClientIsConnected {Poco::Event::EventType::EVENT_MANUALRESET}
+    , mStopHandleSubjectsChangedThread {Poco::Event::EventType::EVENT_AUTORESET}
+{
+}
+
+aos::Error VISIdentifier::Init(std::shared_ptr<aos::iam::identhandler::SubjectsObserverItf> subjectsObserverPtr,
+    const long                                                                              initTimeoutMilliseconds)
+{
+    if (subjectsObserverPtr == nullptr) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eInvalidArgument);
+    }
+
+    std::lock_guard lock(mMutex);
+
+    if (mWsClientPtr == nullptr) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    mSubjectsObserverPtr = std::move(subjectsObserverPtr);
+
+    mHandleConnectionThread = std::thread(&VISIdentifier::HandleConnection, this);
+
+    if (!mWSClientIsConnected.tryWait(initTimeoutMilliseconds)) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eTimeout);
+    }
```

Comment:
why we need in Init method to wait? for my opinion this is not neccessary because our HandleConnection in loop has retry connection. For my opinion this if not necessary

Reply:
Removed timeout logic